v2/internal/plugins/collector/proxy/context.go

```diff
@@ -94,12 +94,8 @@ func (pc *PluginContext) AddMetric(ns string, v interface{}, modifiers ...plugin
 		return fmt.Errorf("invalid value type (%T) for metric: %s", v, ns)
 	}
 
-	parsedNs, err := metrictree.ParseNamespace(ns, false)
-	if err != nil {
-		return fmt.Errorf("invalid format of namespace: %v", err)
-	}
-	if !parsedNs.IsUsableForAddition(pc.ctxManager.metricsDefinition.HasRules(), false) {
-		return fmt.Errorf("invalid namespace (some elements can't be used when adding metric): %v", err)
+	if ok, err := pc.ctxManager.metricsDefinition.IsUsableForAddition(ns, false); !ok || err != nil {
```

Comment:
@adamiklukasz I wonder if we could move some of this logic lower, to tree validator? wdyt?

Reply:
well, it's kind of prechecking if the namespace is valid in the given context. Tree is rather related with definitions from tasks.

Reply:
but this logically belongs to checking metrics namespace vs tree and as such might be blackboxed from here - right now some internals of namespace module implementation leak to here
plus even the call chain clearly shows it belongs to the metricsDefinition layer `pc.ctxManager.metricsDefinition.HasRules()`

Reply:
I mean metrics validation seems more like a responsibility of the tree rather than the context